### ISSUE TITLE
Error with symmetry in PySCF 2.7 pyscf.M(symmetry="C3v")

### ISSUE THREAD

**Issue by abhishekkhedkar-phasecraft** [pyscf/pyscf#2441]
I get an error pyscf.lib.exceptions.PointGroupSymmetryError: Unable to identify input symmetry C3v. Try symmetry="C3v"

Here is my input:
```
from pyscf import scf, M
from pyscf import symm

# Ammonia molecule at equilibrium geometry as an example with expected C3v symmetry
mol=M(atom=' N 0.000 0.000 0.184; H 0.000 0.944 -0.294; H -0.818 -0.472 -0.294; H 0.818 -0.472 -0.294')

print(mol.topgroup, mol.groupname)
# this outputs C1 C1 as expected


mol.build(symmetry=True)
# this outputs Cs Cs, Should it not be C3v ?

symm.geom.TOLERANCE=0.05 # !!!! default is 1e-5

mol.build(symmetry=True)
print(mol.topgroup, mol.groupname)
# this outputs C3v Cs

mol.build(symmetry="C3v")
print(mol.topgroup, mol.groupname)
```

This ^ throws an error:
```
Traceback (most recent call last):
python3.10/site-packages/pyscf/gto/mole.py", line 2651, in _build_symmetry
    groupname, axes = symm.as_subgroup(self.topgroup, axes,
python3.10/site-packages/pyscf/symm/geom.py", line 420, in as_subgroup
    raise PointGroupSymmetryError('%s not in Ablien subgroup of %s' %
pyscf.lib.exceptions.PointGroupSymmetryError: C3v not in Ablien subgroup of C3v

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
input.py", line 11, in <module>
    mol.build(symmetry="C3v")
python3.10/site-packages/pyscf/gto/mole.py", line 2613, in build
    self._build_symmetry()
python3.10/site-packages/pyscf/gto/mole.py", line 2654, in _build_symmetry
    raise PointGroupSymmetryError(
pyscf.lib.exceptions.PointGroupSymmetryError: Unable to identify input symmetry C3v. Try symmetry="C3v"

```

Has how to specify point group symmetry changed in the latest version?

**Comment by matthew-hennefarth** [comment]
> ```python
> mol.build(symmetry=True)
> print(mol.topgroup, mol.groupname)
> # this outputs C3v Cs
> ```

PySCF detects that the point group is C3v, but the closest point group that it can work with the Cs group. I don't believe PySCF supports the C3v point group. The available point groups (and there abelian subgroups) are tabulated [here](https://github.com/pyscf/pyscf/blob/410d9608a73ad3a507b3008a6f68549288ebfc33/pyscf/symm/param.py#L264-L275).

> Has how to specify point group symmetry changed in the latest version?

Were you able to use the C3v subgroup in a prior version of PySCF? If so, can you give the version or commit?

**Comment by abhishekkhedkar-phasecraft** [comment]
Hi Matthew,

Apologies for the late response and thank you so much for pointing that out. I realized that it was always falling back to the corresponding top abelian group in my calculations. I managed to use the msym (libmsym) package to get non-abelian symmetries. In the meantime, I would also like to flag that detection of the symmetry is sensitive to what decimal points geometries being specified in the xyz format. In certain cases relaxing pyscf.symm.geom.TOLERANCE was needed to be able to use symmetries. 

Cheers,
Abhi
